Client/Frontend/Browser/TabManager.swift

```diff
@@ -693,17 +693,27 @@ class TabManagerNavDelegate : NSObject, WKNavigationDelegate {
         completionHandler: (NSURLSessionAuthChallengeDisposition,
         NSURLCredential?) -> Void) {
             var disp: NSURLSessionAuthChallengeDisposition? = nil
+            var canHandleChallenge = false
             for delegate in delegates {
-                delegate.webView?(webView, didReceiveAuthenticationChallenge: challenge) { (disposition, credential) in
-                    // Whoever calls this method first wins. All other calls are ignored.
-                    if disp != nil {
-                        return
+                if delegate.respondsToSelector("webView:didReceiveAuthenticationChallenge:completionHandler:") {
+                    canHandleChallenge = true
+                    delegate.webView?(webView, didReceiveAuthenticationChallenge: challenge) { (disposition, credential) in
+                        // Whoever calls this method first wins. All other calls are ignored.
+                        if disp != nil {
+                            return
+                        }
+
+                        disp = disposition
+                        completionHandler(disposition, credential)
                     }
-
-                    disp = disposition
-                    completionHandler(disposition, credential)
                 }
             }
+            // if we haven't got a delegate that can respond to the challenge, then the completion handler has not been called 
+            // we need to ensure that the completion handler is called otherwise we crash 
+            // so we should return the default handling strategy here
+            if !canHandleChallenge {
+                completionHandler(NSURLSessionAuthChallengeDisposition.PerformDefaultHandling, nil)
+            }
```

Comment:
Nit: I wonder if we can simplify this entire method to:

```swift
    func webView(webView: WKWebView, didReceiveAuthenticationChallenge challenge: NSURLAuthenticationChallenge,
        completionHandler: (NSURLSessionAuthChallengeDisposition,
        NSURLCredential?) -> Void) {

        let callableDelegates = delegates.filter { $0.respondsToSelector("webView:didReceiveAuthenticationChallenge:completionHandler:") }
        guard let delegate = callableDelegates.first else {
            completionHandler(NSURLSessionAuthChallengeDisposition.PerformDefaultHandling, nil)
            return
        }
        delegate.webView?(webView, didReceiveAuthenticationChallenge: challenge, completionHandler: completionHandler)
    }
```

This would filter out any delegates that can't be called and has the check to make sure we call the default handler.